infra/src/main/kotlin/fi/fta/geoviite/api/tracklayout/v1/ExtLocationTrackGeometryServiceV1.kt

```diff
@@ -42,67 +53,240 @@ constructor(
     }
 
+    fun getExtLocationTrackGeometryModifications(
+        oid: Oid<LocationTrack>,
+        trackLayoutVersionFrom: Uuid<Publication>,
+        trackLayoutVersionTo: Uuid<Publication>?,
+        extResolution: ExtResolutionV1?,
+        coordinateSystem: Srid?,
+        addressFilterStart: ExtMaybeTrackKmOrTrackMeterV1?,
+        addressFilterEnd: ExtMaybeTrackKmOrTrackMeterV1?,
+    ): ExtLocationTrackModifiedGeometryResponseV1? {
+        val publications = publicationService.getPublicationsToCompare(trackLayoutVersionFrom, trackLayoutVersionTo)
+        // Lookup before change check to produce consistent error if oid is not found
+        val id = idLookup(locationTrackDao, oid)
+        val resolution = extResolution?.toResolution() ?: Resolution.ONE_METER
+        val coordinateSystem = coordinateSystem ?: LAYOUT_SRID
+        val addressFilter = createAddressFilter(addressFilterStart, addressFilterEnd)
+        return if (publications.areDifferent()) {
+            createGeometryModificationResponse(oid, id, publications, resolution, coordinateSystem, addressFilter)
+        } else {
+            publicationsAreTheSame(trackLayoutVersionFrom)
+        }
+    }
+
+    private fun createGeometryModificationResponse(
+        oid: Oid<LocationTrack>,
+        id: IntId<LocationTrack>,
+        publications: PublicationComparison,
+        resolution: Resolution,
+        coordinateSystem: Srid,
+        addressFilter: AddressFilter,
+    ): ExtLocationTrackModifiedGeometryResponseV1? {
+        val branch = publications.to.layoutBranch.branch
+        val startMoment = publications.from.publicationTime
+        val endMoment = publications.to.publicationTime
+        return publicationDao
+            .fetchPublishedLocationTrackGeomsBetween(id, startMoment, endMoment)
+            ?.let { (old, new) -> old?.let(locationTrackDao::fetch) to locationTrackDao.fetch(new) }
+            ?.takeIf { (old, new) -> old?.exists == true || new.exists }
+            ?.let { (oldTrack, newTrack) ->
+                val oldPoints =
+                    oldTrack
+                        ?.takeIf { it.exists }
+                        ?.let { track -> getAddressPoints(branch, startMoment, track, resolution, addressFilter) }
+                val newPoints =
+                    newTrack
+                        .takeIf { it.exists }
+                        ?.let { track -> getAddressPoints(branch, endMoment, track, resolution, addressFilter) }
+                ExtLocationTrackModifiedGeometryResponseV1(
+                    trackLayoutVersionFrom = publications.from.uuid,
+                    trackLayoutVersionTo = publications.to.uuid,
+                    locationTrackOid = oid,
+                    coordinateSystem = coordinateSystem,
+                    trackIntervals = createModifiedTrackIntervals(oldPoints, newPoints, coordinateSystem),
+                )
+            }
+    }
+
+    private fun createModifiedTrackIntervals(
```

Comment:
Tässä tämä varsinainen diffailufunktio. Tämä on rumuudestaan huolimatta puhdas funkkari ja tarkoitus on että kun toteutan saman setin ratanumerolle niin siirrän tämän johonkin omaan tiedostoonsa. Sille voi myös kirjoittaa siinä kohtaa helpommin erikoiscase-testejä (api-testit todentaa tämän kyllä pääpiirteittäin, mutta kulma-caseja ei kannattane testata sillä tasolla). 

Ehkä sen erottelun voisi tietty tehdä jo nytkin, mutta voisin ottaa tästä versiosta ensin pullarin ja todentaa tuon suorituskykyä vähän jne ensin.